src/Agent/NewRelic/Agent/Core/Api/AgentApiImplementation.cs

```diff
@@ -519,7 +519,42 @@ public string GetBrowserTimingHeader()
                 // The transaction's name must be frozen if we're going to generate a RUM script
                 transaction.CandidateTransactionName.Freeze(TransactionNameFreezeReason.ManualBrowserScriptInjection);
 
-                return _browserMonitoringScriptMaker.GetScript(transaction) ?? string.Empty;
+                return _browserMonitoringScriptMaker.GetScript(transaction, null) ?? string.Empty;
+            }
```

Comment:
Just a small nitpicking, the logic of the `public string GetBrowserTimingHeader()` can be greatly simplified by doing
```
public string GetBrowserTimingHeader()
{
   GetBrowserTimingHeader(null);
}
```
we like remove duplicate code a much as possible.

Reply:
I can do that, but then were' back to `GetBrowserTimingHeader(string? nonce)` (for nullability) which you initially objected to, and was why I introduced an entirely new overload. Maybe you were just asking for the internal API to match the public API?

The method now simply delegates to the other implementation.

Reply:
@bgrainger as a good practice, we are very careful with using nullable parameters as method arguments. Imagine, for some reasons, we would introduce a new overload like `GetBrowserTimingHeader(NewType? nonce)`. We now have 2 apis as the followings.

`GetBrowserTimingHeader(SomeType? nonce)`
`GetBrowserTimingHeader(string? nonce)`

What happens when a user call `GetBrowserTimingHeader(null)`. The user will get a compile error.

Reply:
That will happen whether or not it's written `string` or `string?`, since `string` is a reference type and can always accept `null` as an argument value. This change is _purely_ for compile-time nullability flow analysis.

If you like, I can change it back to `string` and use `default!` in the call site; it's somewhat irrelevant from an ABI perspective.

Reply:
Would you like this renamed to `GetBrowserTimingHeaderWithNonce` to avoid future overloaded API issues?

I'm not at all committed to the current name choice; as indicated in the OP this is just a suggestion for the new API to get the process started.

Reply:
> That will happen whether or not it's written `string` or `string?`, since `string` is a reference type and can always accept `null` as an argument value. This change is _purely_ for compile-time nullability flow analysis.
> 
> If you like, I can change it back to `string` and use `default!` in the call site; it's somewhat irrelevant from an ABI perspective.

You are absolutely right. I realized my concerns about optional parameter were totally off. I want to take a step back before I overcomplicate this myself. I just recently updated your branch with changes. Please do let me know if they look ok to you.